portal/common/lib/page_fetching.ts

```diff
@@ -78,6 +83,19 @@ export async function fetchPage(
 
     // Deserialize the bcs encoded body and decompress.
     const body = await contents.arrayBuffer();
+    // Verify the integrity of the aggregator response by hashing
+    // the response contents.
+    const h10b = sha256(
+        new Uint8Array(body),
+    );
+    if (result.blob_hash != toB64(h10b)) {
+        console.warn(
+            '[!] checksum mismatch [!] for:', result.path, '.',
+            `blob hash: ${result.blob_hash} | aggr. hash: ${toB64(h10b)}`
+        )
+        return generateChecksumErrorResponse()
+    }
+
     const decompressed = await decompressData(new Uint8Array(body), result.content_encoding);
```

Comment:
Important question: here you check the hash before decompression. Is this the intended way. Note that this choice will also influence the site builder.

advantages: hashing a compressed string can be faster.
disadvantages: this hash does not match the SRI hash anymore.

Reply:
Hmm that's true, I think I would go with checking the hash _after_ decompression.